Specify precisely where it is an error to declare an initializing formal (#3517)
dart-lang/language: specification/dartLangSpec.tex

It is a compile-time error if \id{} is not also the name of
an instance variable of the immediately enclosing class or enum.

\LMHash{}%
It is a compile-time error for an initializing formal parameter
to occur in any function which is not a generative constructor.
Also, it is a compile-time error for an initializing formal parameter

> to occur in a redirecting and in an external constructor.

Reviewer: [optional] 

```suggestion
to occur in a redirecting or external constructor.
```
